api/bot/slackbot/v1/slackbot.proto

```diff
@@ -0,0 +1,113 @@
+syntax = "proto3";
+
+package clutch.bot.slackbot.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/bot/slackbot/v1;slackbotv1";
+
+import "api/v1/annotations.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+service SlackBotAPI {
+  rpc Event(EventRequest) returns (EventResponse) {
+    option (google.api.http) = {
+      post : "/v1/bot/slackbot/event"
+      body : "*"
+    };
+    option (clutch.api.v1.action).type = CREATE;
+  }
+}
+
+// Information on the bot user
+message Bot {
+  // bot id
+  string id = 1;
+  // if the bot is deleted
+  bool deleted = 2;
+  // bot name
+  string name = 3;
+  // timestamp of when the bot app was last updated
+  int64 updated = 4;
+  // unique identifier of the installed Slack application
+  string app_id = 5;
+  // image urls of the bot's display picture
+  map<string, string> icons = 6;
+  // unique identifier of the workspace where the event occurred
+  string team_id = 7;
+}
+
+// The structure of events vary among types. Full list of event types: https://api.slack.com/events.
+// We will be receiving app_mention events (events that mention the bot) or message.im events (a message posted in a DM
+// with the bot).
+message Event {
+  // the type of event
+  string type = 1;
+  // the user id of the user who messaged the bot
+  string user = 2;
+  string bot_id = 3;
+  // metadata on the bot
+  Bot bot_profile = 4;
```

Comment:
if we have the `bot_profile` do we need the `bot_id` as a top level key?

Reply:
yeah, the `bot_profile` only gets sent when you DM the bot directly (it's an "im" event and the Slack API sends us some extra metadata) so for other event types, we wouldn't get this and can still make use of the `bot_id` top level key